clients/drcachesim/tools/func_view.cpp

```diff
@@ -170,8 +171,13 @@ func_view_t::process_memref_for_markers(void *shard_data, const memref_t &memref
         shard->last_was_syscall = memref.marker.marker_value >=
             static_cast<int64_t>(func_trace_t::TRACE_FUNC_ID_SYSCALL_BASE);
     }
-    if (shard->last_was_syscall)
+    // OFFLINE_FILE_TYPE_ARCH_REGDEPS traces have only SYS_futex function-related (i.e.,
+    // TRACE_MARKER_TYPE_FUNC_) markers for which we do want to print "returns"
+    // statistics, so we disable this check for these traces.
```

Comment:
Not sure I understand: if we want statistics on system calls whose parameters were traced, shouldn't we enable those stats for all system calls for all traces?  Why enable only for ARCH_REGDEPS?

Reply:
I was preserving the previous behavior of the tool.
I assumed that func_view was intended to print "user-level functions", hence the absence of syscalls statistics (which is hard coded here).
I thought we wanted to make an exception for REGDEPS traces only, since the only "function" there is the SYS_futex syscall.
I'm not against enabling syscall stats for all types traces (as long as there is an entry in funclist.log) though.
Should we do that?